docs/src/build_tips.md

```diff
@@ -0,0 +1,135 @@
+
+* [staticfloat/IpoptBuilder](https://github.com/staticfloat/IpoptBuilder/blob/master/build_tarballs.jl)
+
+* [davidanthoff/SnappyBuilder](https://github.com/davidanthoff/SnappyBuilder/blob/master/build_tarballs.jl)
+
+* [JuliaDiffEq/SundialsBuilder](https://github.com/JuliaDiffEq/SundialsBuilder/blob/master/build_tarballs.jl)
+  - Needs `-DSUNDIALS_INDEX_TYPE=int32_t` on 32-bit targets 
+  - Needs to copy *.dll files from `destdir/lib` to `destdir/bin` for windows; this also removes symlinks by using `cp -L`
+  - Needs `-DCMAKE_FIND_ROOT_PATH="$WORKSPACE/destdir"`, so CMake's `find_library` can find libraries from KLU
+
+
+## Builds with binary dependencies
+
+A build script can depend on binaries generated by another Builder repository. A builder specifies `dependencies` like:
+
+```julia
+dependencies = [
+    # We need libogg to build FLAC
+    "https://github.com/staticfloat/OggBuilder/releases/download/v1.3.3-0/build.jl"
+]
+```
+
+Each of the `dependencies` points to a `build.jl` file, usually provided with a release of another Builder repository.
+
+In the wizard, this can be specified with the prompt: *Do you require any (binary) dependencies?  [y/N]*.
+
+Examples of builders that depend on other binaries include:
+
+* [staticfloat/FLACBuilder](https://github.com/staticfloat/FLACBuilder/blob/master/build_tarballs.jl) depends on [staticfloat/OggBuilder](https://github.com/staticfloat/OggBuilder/blob/master/build_tarballs.jl) ([build.jl](https://github.com/staticfloat/OggBuilder/releases/download/v1.3.3-0/build.jl)).
+
+
+## Editing files in the wizard
+
+In the wizard, the `vi` editor is available for editing files. But, it doesn't leave any record in the build script. One generally needs to provide patch files or sue something like `sed`. Here is an approach using `diff` and `patch`:
+
+```sh
+cp file.ext file.ext.orig
+vi file.ext     # make the changes
+diff -u file.ext.orig file.ext
+# Create a patch based on the results copy-pasted from the output of `diff`
+cat > file.patch <<'END'
+--- file.ext.orig 2017-12-14 19:28:48.816021000 -0500
++++ file.ext2017-12-14 19:29:03.912021000 -0500
+@@ -1,4 +1,5 @@
+ -https://computation.llnl.gov/projects/sundials/download/sundials-3.0.0.tar.gz
+ +https://computation.llnl.gov/projects/sundials/download/sundials-3.1.0.tar.gz
+  
+  http://faculty.cse.tamu.edu/davis/SuiteSparse/SuiteSparse-5.0.0.tar.gz
+ 
+END
+# Apply the patch
+patch -l file.ext.orig file.patch -o file.ext
+```
+
+There are plans to handle file changes in the wizard automatically [(#25)](https://github.com/JuliaPackaging/BinaryBuilder.jl/issues/125).
+
+
+## Other examples
+
+Examples of other interesting builders include:
+
+* [staticfloat/IpoptBuilder](https://github.com/staticfloat/IpoptBuilder/blob/master/build_tarballs.jl) -- Includes multiple sources, and each of these has different build approaches. Includes custom `CC` and `CXX` definitions for macOS to use Clang and sets `LDFLAGS` to specify a minimum macOS version. 
```

Comment:
The special things here for `IpoptBuilder` are about to go away, sorry!  I promise we'll have more unusual things for you to include here soon.  :)